libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/certificates/DefaultUserCertificatesProvider.kt

```diff
@@ -37,41 +38,54 @@ class DefaultUserCertificatesProvider @Inject constructor() : UserCertificatesPr
      *         in encoded form.
      */
     override fun provides(): List<ByteArray> {
-        val certs = mutableListOf<ByteArray>()
-
         // At least for API 34 the `AndroidCAStore` `Keystore` type contained user certificates as well.
         // I have not found this to be documented anywhere.
-        val keyStore: KeyStore = KeyStore.getInstance("AndroidCAStore").apply {
-            load(null)
+        val keyStore: KeyStore = try {
+            KeyStore.getInstance("AndroidCAStore")
+        } catch (e: KeyStoreException) {
+            Timber.w(e, "Failed to get AndroidCAStore keystore")
+            return emptyList()
         }
-
-        val aliases = keyStore.aliases()
-
-        while (aliases.hasMoreElements()) {
-            val alias = aliases.nextElement()
-            val entry = keyStore.getEntry(alias, null)
-
-            if (entry is KeyStore.TrustedCertificateEntry) {
+        try {
+            keyStore.load(null)
+        } catch (e: Exception) {
+            Timber.w(e, "Failed to load AndroidCAStore keystore")
+            return emptyList()
+        }
+        val aliases = try {
+            keyStore.aliases()
+        } catch (e: Exception) {
+            Timber.w(e, "Failed to get aliases from AndroidCAStore keystore")
```

Comment:
Couldn't you merge the first 2 blocks? I think the third could be merged too if we use different catch blocks.

Reply:
I wanted to have specific log to eventually investigate what was failing. But probably the Exception will contain enough details. WDYT?

Reply:
Yeah, I think so. Also, the stacktrace should point at which function threw it.

Reply:
9fed8689682dacd9382030ce766f210ff479e608

Can't really merge the first block, since `keyStore` is used later to get the entry.